### PR TITLE
feat(jans-cedarling): introduce batch authorize types and extract setup helpers

### LINKED ISSUE #14489: feat(jans-cedarling): introduce batch authorize request/response types
**Is your feature request related to a problem? Please describe.**

Before implementing batch authorization logic, we need stable request and
response shapes that both flows (unsigned and multi-issuer) can share. Landing
the types first lets subsequent core-engine and binding PRs be reviewed against
a fixed contract.

**Describe the solution you'd like**

Add three public types plus their validation:

- `BatchItem` — carries `resource: EntityData`, `action: String`,
  `context: serde_json::Value`. Shared between both flows.
- `BatchAuthorizeUnsignedRequest` — `principal: Option<EntityData>`, `items:
  Vec<BatchItem>`.
- `BatchAuthorizeMultiIssuerRequest` — `tokens: Vec<TokenInput>`, `items:
  Vec<BatchItem>`.

Validation rules (return a new `BatchValidationError` variant per failure):

- `items` empty → error.
- Any item with a non-object `context` value → error, matching the existing
  `MultiIssuerValidationError::InvalidContextJson` behavior.
- For the multi-issuer request, `tokens` empty → error, matching the current
  `AuthorizeMultiIssuerRequest::validate` behavior.

Add unit tests covering: valid construction, empty-items rejection,
non-object-context rejection, JSON round-trip serialization.

**Describe alternatives you've considered**
N/A

**Additional context**

This issue lands types only. No engine method is added, no binding is touched.
It is a small, self-contained PR whose diff is easy to review against the master
issue's contract.

### PR DESCRIPTION
**Pull request by haileyesus2433**
### Prepare

- [x] Read [PR guidelines](https://github.com/JanssenProject/jans/blob/main/docs/CONTRIBUTING.md#prs)
- [x] Read [license information](https://github.com/JanssenProject/jans/blob/main/LICENSE)

-------------------

#### Target issue
  
  
  
  
closes #14489 #14490

#### Implementation Details
  

-------------------
### Test and Document the changes
- [x] Static code analysis has been run locally and issues have been fixed
- [x] Relevant unit and integration tests have been added/updated
- [ ] Relevant documentation has been updated if any (i.e. user guides, installation and configuration guides, technical design docs etc)

Please check the below before submitting your PR. The PR will not be merged if there are no commits that start with `docs:` to indicate documentation changes or if the below checklist is not selected.
- [x] **I confirm that there is no impact on the docs due to the code changes in this PR.**

### CODE REVIEW

**Comment by mo-auto** [comment]
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scan Engine |  Critical |  High |  Medium |  Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/janssenproject/pr-checks/318070a2-4b9c-4b7a-a024-dbd48f7b9c30?refs=d183d659bc1223e7b7ade24cc2c88a7c62d7332a&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

File: jans-cedarling/cedarling/src/authz/request.rs [hunk 1]
```diff
@@ -147,6 +148,126 @@ impl AuthorizeMultiIssuerRequest {
...
+/// enclosing batch request.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct BatchItem {
+    /// Resource being accessed for this item.
+    pub resource: EntityData,
+    /// Action being performed on this item.
+    pub action: String,
+    /// Per-item context (JSON object). Non-object values are rejected at
+    /// validate-time to match `RequestUnsigned::context` semantics.
+    pub context: Value,
```
Comment (dagregi) [inline jans-cedarling/cedarling/src/authz/request.rs:164]: context for single multi-issuer request is optional should this be too?

Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/authz/request.rs:164]: Since BatchItem is shared between both batch flows we have to pick one shape. I went with Value to match RequestUnsigned, but I think what you're actually pointing at is the JSON ergonomics callers of the multi-issuer single-item API can omit context entirely today, and losing that under batch would be an awkward step down.

I'd rather not change the Rust type to Option<Value>: build_context and build_multi_issuer_context both take owned Value, so Option would force .unwrap_or(json!({})) at every call site and add a None variant that has to be threaded through the downstream helpers.

i will keep the Rust type as Value but make it default-omittable in serde

Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/authz/request.rs:164]: it will be added in the next PR

Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/authz/request.rs:164]: it will be added in the next PR

File: jans-cedarling/cedarling/src/authz/errors.rs [hunk 2]
```diff
@@ -44,6 +44,22 @@ pub enum TokenInputError {
...
+    #[error("Empty items array")]
+    EmptyItems,
+
+    /// The `tokens` array was empty (multi-issuer batch only).
+    #[error("Empty tokens array")]
+    EmptyTokens,
+
+    /// A batch item's context field is not a JSON object.
+    #[error("Context for item {index} must be a JSON object")]
+    InvalidItemContext { index: usize },
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/authz/errors.rs:60]: I think we can make `index` public

File: jans-cedarling/cedarling/src/entity_builder/build_multi_issuer_entity.rs [hunk 3]
```diff
@@ -240,17 +241,68 @@ fn determine_token_entity_type(token: &Token) -> String {
...
+            tokens: setup.tokens,
+            resource,
+            default_entities: setup.default_entities,
+        })
+    }
+
+    /// Build the resource-independent multi-issuer entities (tokens + issuers +
+    /// default entities). Used by batch authorization to amortize entity
+    /// construction across items and by the single-item path via
+    /// [`Self::build_multi_issuer_entities`].
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/entity_builder/build_multi_issuer_entity.rs:298]: now the `build_multi_issuer_entities` is unused, is it intentional? If yes maybe we should remove it?

File: jans-cedarling/cedarling/src/authz/request.rs [hunk 4]
```diff
@@ -147,6 +148,126 @@ impl AuthorizeMultiIssuerRequest {
...
+    pub fn new(tokens: Vec<TokenInput>, items: Vec<BatchItem>) -> Self {
+        Self { tokens, items }
+    }
+
+    /// Validate the batch request structure.
+    ///
+    /// Returns [`BatchValidationError::EmptyTokens`] if `tokens` is empty,
+    /// [`BatchValidationError::EmptyItems`] if `items` is empty, and
+    /// [`BatchValidationError::InvalidItemContext`] with the offending index
+    /// if any item's `context` is not a JSON object.
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/authz/request.rs:231]: We have docstring in error so don't need to duplicate

File: jans-cedarling/cedarling/src/authz/request.rs [hunk 5]
```diff
@@ -147,6 +148,126 @@ impl AuthorizeMultiIssuerRequest {
...
+    pub fn validate(&self) -> Result<(), BatchValidationError> {
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/authz/request.rs:232]: This method is used only in tests

File: jans-cedarling/cedarling/src/entity_builder/mod.rs [hunk 6]
```diff
@@ -94,29 +93,25 @@ impl EntityBuilder {
...
+    ///
+    /// Callers that need to amortize principal construction across many items
+    /// (batch authorization) call this once and then invoke
+    /// [`Self::build_resource_entity`] per item.
+    pub(crate) fn build_unsigned_principal(
         &self,
-        request: &RequestUnsigned,
-    ) -> Result<BuiltEntitiesUnsigned, BuildUnsignedEntityError> {
+        principal: Option<&EntityData>,
+    ) -> Result<(Option<Entity>, BuiltEntities), BuildUnsignedEntityError> {
```
Comment (olehbozhok) [inline jans-cedarling/cedarling/src/entity_builder/mod.rs:104]: Please, instead of returning `(Option<Entity>, BuiltEntities)` create a struct with meaningful attributes names.

**Review by olehbozhok — CHANGES_REQUESTED** [review]
I would like to avoid haiving new `#[allow(dead_code)]` . And I don't really understand why we don't have public methods for batch execution in `cedarling/src/lib.rs`

**Review by olehbozhok — APPROVED** [review]

File: jans-cedarling/cedarling/src/authz/errors.rs (hunk 2 above)
Comment (haileyesus2433) [reply jans-cedarling/cedarling/src/authz/errors.rs:60]: the `#[allow(dead_code)]` is until things come together i will remove them as i go along,  Enum-variant struct fields inherit visibility from the enum itself, so index is already public

**Comment by haileyesus2433** [comment]
> I would like to avoid haiving new `#[allow(dead_code)]` . And I don't really understand why we don't have public methods for batch execution in `cedarling/src/lib.rs`

all of them will be addressed in the coming PR this is just for laying the foundation and dividing things up so that its easier to review
